src/migrations/merge.rs

```diff
@@ -0,0 +1,172 @@
+use crate::migrations::migration::MigrationFile;
+use crate::migrations::rebase::{fix_migration_ids};
+use crate::print;
+
+pub struct MergeMigrations {
+    pub base_migrations: IndexMap<String, MergeMigration>,
+    pub target_migrations: IndexMap<String, MergeMigration>
+}
+
+impl MergeMigrations {
+    fn flatten(&self) -> IndexMap<&String, &MergeMigration> {
+        let mut result = IndexMap::new();
+
+        for migration in self.base_migrations.iter() {
+            result.insert(migration.0, migration.1);
+        }
+
+        for migration in self.target_migrations.iter() {
+            result.insert(migration.0, migration.1);
+        }
+
+        result
+    }
+}
+
+pub struct MergeMigration {
+    key: MigrationKey,
+    id_override: Option<String>,
+    migration: DBMigration,
+    parent_override: Option<String>
+}
+
+impl MigrationToText for MergeMigration {
+    type StatementsIter<'a> = std::iter::Once<&'a String> where Self: 'a;
+
+    fn key(&self) -> &MigrationKey {
+        &self.key
+    }
+
+    fn parent(&self) -> anyhow::Result<&str> {
+        Ok(self.parent_override.as_ref().or(self.migration.parent_names.first()).map(|v| v.as_str()).unwrap_or("initial"))
+    }
+
+    fn id(&self) -> anyhow::Result<&str> {
+        Ok(self.id_override.as_ref().unwrap_or(&self.migration.name))
+    }
+
+    fn statements<'a>(&'a self) -> Self::StatementsIter<'a> {
+        std::iter::once(&self.migration.script)
+    }
+}
+
+pub async fn get_merge_migrations(base: &mut Connection, target: &mut Connection) -> anyhow::Result<MergeMigrations> {
+    let base_migrations = read_all(base, true, false).await?;
+    let mut target_migrations = read_all(target, true, false).await?;
+
+    if base_migrations.len() > target_migrations.len() {
+        anyhow::bail!("Source branch contains more migrations than the target branch");
+    }
+
+    let mut target_missing = Vec::new();
+    let mut diverging_migrations = Vec::new();
+
+    for (index, (base_migration_id, _)) in base_migrations.iter().enumerate() {
+        if let Some((target_migration_id, _)) = target_migrations.get_index(index) {
+            if target_migration_id != base_migration_id {
+                diverging_migrations.push((index, base_migration_id, target_migration_id))
+            }
+        } else {
+            target_missing.push((index, base_migration_id))
+        }
+    }
+
+    if diverging_migrations.len() > 0 {
+        eprintln!("\nThe migration history of {} diverges from {}:", target.database(), base.database());
+
+        for (index, expecting, actual) in &diverging_migrations {
+            eprintln!("{}. Expecting {} but has {}", index, &expecting[..7], &actual[..7])
+        }
+    }
+
+    if target_missing.len() > 0 {
+        eprintln!("\nThe migration history of {} is missing the following migrations", target.database());
+
+        for (index, missing) in &target_missing {
+            eprintln!("{}. {}", index, &missing[..7])
+        }
+    }
```

Comment:
Hmm, in what case will this happen?

Reply:
This happens when the target branch doesn't contain a specific migration; ex:
**Base Branch**
- A
- B
- C

**Feature**
- A
- B

The message will appear with missing migration of C

Reply:
Well, this is ok - it's a no-op: there is nothing from feature branch to merge into base. Git says:

```
➜  edgedb git:(refactor-parser-context) git merge HEAD~1
Already up to date.
```